lib/improver/metadata/forecast_times.py

```diff
@@ -70,10 +72,13 @@ def forecast_period_coord(
             DimCoord and this coord does not need changing, otherwise
             it will be an AuxCoord. Units are result_units.
     """
+    create_dim_coord = False
     if cube.coords("forecast_period"):
         fp_type = cube.coord("forecast_period").dtype
+        if isinstance(cube.coord("forecast_period"), iris.coords.DimCoord):
+            create_dim_coord = True
     else:
-        fp_type = np.int32
+        fp_type = TIME_INTERVAL_DTYPE
```

Comment:
Should you say in the docstring what is used if the result_units is not supplied?

Reply:
I should probably make the function not return anything not np.int32.  But this was a quick refactor to make Codacy happy.  Perhaps I could make improvements to this in an immediate follow-on PR?

Reply:
Also I should probably not say in the docstring what's returned explicitly - that's sort of the point of centralising the expected types, so that we don't have to update that information (code or docstring alike) everywhere.